server/service/src/stocktake_line/validate.rs

```diff
@@ -99,3 +100,28 @@ pub fn check_reason_is_valid(
     }
     Ok(false)
 }
+
+pub fn check_stock_line_reduced_below_zero(
```

Comment:
Logic here should be:

total (stock take line) - counted (stock take line) = adjustment

if adjustment > 0 AND total (stock line) - adjustment < 0 or available (stock line) - adjustment < 0 {
  error
}

Reply:
Will do all the changes https://github.com/openmsupply/open-msupply/issues/1185 here and assign you as a reviewer (: